---
title: "Module 3: Project 1 by Team 5"
subtitle: "mothur vs. QIIME2 Microbiome Data Analysis"
author: 
  - Karl Abuan
  - May Ho
  - Jonah Lin
  - Leilynaz Malekafzali
  - Tiffany Yang
  - Abdur Rahman M. A. Basher
abstract: |
  This is the abstract.
  It consists of two paragraphs.
date: "`r format(Sys.time(), '%B %d, %Y')`"
header-includes:
   - \newcommand*{\secref}[1]{Section~\ref{#1}}
output:
  pdf_document:
    toc: yes
    toc_depth: 3
    number_sections: yes
    fig_caption: true
  html_document:
    toc: yes
    toc_float:
      collapsed: no
mainfont: Times
fontsize: 11 pt
editor_options:
  chunk_output_type: console
urlcolor: blue
link-citations: yes
bibliography: main.bib
---

```{r global_options, include=FALSE}
knitr::clean_cache(clean = TRUE)
knitr::opts_chunk$set(fig.width=8, fig.height=6, fig.align = "center", echo=FALSE, warning=FALSE, include=TRUE, tidy = TRUE, message=FALSE)
```


```{r, eval=FALSE}
## try http:// if https:// URLs are not supported
source("https://bioconductor.org/biocLite.R")  
biocLite("phyloseq")
library("tidyverse")
library("gridExtra")
library("magrittr")
library("ggpubr")
library("RColorBrewer")

```


```{r, include=FALSE}
source("https://bioconductor.org/biocLite.R")  
library("phyloseq")
library("tidyverse")
library("gridExtra")
library("magrittr")
library("ggpubr")
library("knitr")
library("RColorBrewer")

# theme_set(theme_bw())
# pal = "Set1"
# scale_colour_discrete <-  function(palname=pal, ...){
#   scale_colour_brewer(palette=palname, ...)
# }
# scale_fill_discrete <-  function(palname=pal, ...){
#   scale_fill_brewer(palette=palname, ...)
# }
```
* * *
# Introduction

[Talk about microbial community deriving the most biogeochemical forces with citations]. The wide diversity of microbes present in the different ecosystem implies that an almost infinite number of individuals needs to be identified to accurately describe such communities. Nevertheless, the advancements in analyzing high-throughput marker-gene sequencing data made it feasible to construct molecular operational taxonomic units (OTUs) through clustering the sequencing reads using a variety of dissimilarity distance methods [@Chen2013:comparison]. Another compelling approach to cluster group of reads is based on amplicon sequence variants (ASVs) [@Callahan2017:exact]. 

We resort to studying microbial composition obtained from Saanich Inlet oxygen minimum zone (OMZ). Saanich Inlet is a seasonally anoxic fjord on the coast of Vancouver Island, British Columbia, Canada [@Hawley2017:compendium;@Torres2017:compendium]. Most of the year, the fjord has an anoxic basin. At the end of summer, oxic waters flow into the basin, resulting in renewing the oxygen. From an OMZ research perspective, the Saanich Inlet is considered important as it provides an opportunity to study microbial ecology and various nutrient cycling in OMZs  particularly under oxic-anoxic shifts [@Hallam2017:monitoring] [You might add additional information]

Planctomyces is .... [Why did you choose this genus why is it important to study?]

To this end, we examined microbial diversity in Saanich Inlet dataset that was preprocess using mothur[@Schloss2009:introducing] and QIIME2[I didn't find citation]. While mothur uses [Talk about OTU here], QIIME2 produces ASVs [give a brief description about ASVs]. [Talk about differences between these two methods].

We further address the planctomyces diversity correlation with oxygen and depth. We show that under the statistical framework there is little evidence to support our initial hypothesis. However, we claim that the linear model was not adequate to provide insightful knowledge regarding the existence of such correlations.

The remaining of the paper is organized as follows. Section 2 describes the problem statements. Followed by [FILL] in Section 3. Finally, we summarize our contributions in Section 4.

# Problem Formulation
[Talk in depth about Planctomyces]

# Materials and Experimental Configuration
## Experimental Protocols

To understand the correlation of microbial diversity and oxygen concentration across samples, we report four experimentally designed test protocols:

**P1.** Analysis of microbial community structure along with depth and oxygen concentration.

**P2.** Analysis of abundance information of Planctomyces along with depth and/or oxygen concentration.

**P3.** Estimate richness (number of OTUs/ASVs) for Planctomyces.

**P4.** Interpretation of abundance information of OTUs/ASVs of Planctomyces along with depth and/or oxygen concentration.


## Dataset
[Talk about Saanich Inlet dataset and various properties as documented in [@Hawley2017:compendium;@Torres2017:compendium]]

## Methods

### Shannon Diversity Index (SDI) and Chao1
We applied _Shannon diversity index (SDI)_ to estimate the microbial diversity of  Saanich Inlet dataset. It has the following definition: 

$SDI = -\sum_i^R p_i \log(p_i)$

where $p_i$ represents the distribution of individuals belonging to the $i$th species, and $R$ represents the number of distinct species [@Finotello2016:measuring]. It can be noted that SDI takes both the richness and abundance information to measure the expected uncertainty about species contained in a sample. The high SDI value suggests that species are evenly distributed while low  SDI value implies species are disproportionality situated. SDI value could be zero meaning the sample contains exactly one or no species at all. However, SDI does not directly model the expected richness of a sample and, neither, it represents an accurate estimation of species diversity because the probability distribution of species is not knowable exactly; it is only an estimate from a sample.

In contrast to SDI, _Chao1_ could be used to recover approximate true richness:

$Chao1=S_{obs}+\frac{\alpha}{2\beta}$

where $S_{obs}$ represents the observed richness, $\alpha$ and $\beta$ indicates the number of different species with exactly one or more than two counts, respectively. The Chao1 method is used to rectify the richness by including the distribution of the rarest species [@Finotello2016:measuring].

### General Linear Model

General linear model (LM) [@Hastie2009:elements] is employed to recover interactions between several factors that might be exhibited in Saanich Inlet dataset. In our experiments, we use a single regression model that relates a dependent variable $y$ (abundance) to a single quantitative independent variable $x_1$ (depth or oxygen), and it has the following form:

$y=\theta_0+\theta_1x_1+\epsilon$

The parameter $\theta_0$ is the $y$-intercept, which represents the expected value of $y$ when $x_1$ is zero. The parameter $\theta_1$ is the slope of the regression line, and it represents the expected change (positive or negative) in $y$ (abundance) for a unit increase in $x_1$ (depth or oxygen). $\theta_1$ could be $0$ indicating no effective change with $x_1$. And, $\epsilon$ is the error term and is usually set to $0$.

All the parameters could be estimated using ordinary least squares (OLS). However, to test the significance $\theta_1$, we formulate hypothesis testing: i)- the null hypothesis $H_0:\theta_1>\gamma$, which asserts that no additional predictive value over and above, contributed by $\theta_1$ and the $\gamma$ is an arbitrary cutoff probability from t-student distribution ; ii)- the alternative hypothesis $H_1: \theta_1\leq \gamma$ measures whether $x_j$ has additional predictive strengths.

If the weight of $\theta_1$, referred to as the level of significance (or $p$-value) and defines as a probability, is below or equal to $\gamma$ then we accept $H_1$; otherwise, accept $H_0$.


## Data Preporocessing

```{r loading}
load("data/mothur_phyloseq.RData")
load("data/qiime2_phyloseq.RData")
set.seed(4832)
rarefiedM <- rarefy_even_depth(mothur, sample.size=100000)
rarefiedQ <- rarefy_even_depth(qiime2, sample.size=100000)
rarefiedMPer = transform_sample_counts(rarefiedM, function(x) 100 * x/sum(x))
rarefiedQPer = transform_sample_counts(rarefiedQ, function(x) 100 * x/sum(x))
```

```{r preprocessing}
# First rule
firstMTaxa <- filter_taxa(rarefiedMPer, function(x) sum(x == 0) <= 4, TRUE)
firstQTaxa <- filter_taxa(rarefiedQPer, function(x) sum(x == 0) <= 4, TRUE)

# Second rule
basedOnGenus <- as.data.frame(tax_table(firstMTaxa)) %>% 
  filter(!str_detect(Genus, 'uncultured|unclassified'))
secondMTaxa = subset_taxa(firstMTaxa, Genus %in% basedOnGenus$Genus)

basedOnGenus <- as.data.frame(tax_table(firstQTaxa)) %>%
  filter(!str_detect(Genus, 'uncultured|unclassified|\\bD_5__\\b'))
secondQTaxa <- subset_taxa(firstQTaxa, Genus %in% basedOnGenus$Genus)
```

The samples were rarefied/normalized to $100,000$ sequences per sample to facilitate comparisons between samples. The rarefied counts were then converted to relative abundance percentages. Next, we perform a series of filterings according to three rules: i)- exclude OTUs that are not observed for more than $4$ samples, and ii)- prune samples and OTUs with unknown values, such as ``unclassified`` value. This has resulted in `r ntaxa(secondMTaxa)` and `r ntaxa(secondQTaxa)` taxa from mothur and QIIME2, respectively. No other preprocessing were applied. The implementations are done entirely using R (v 3.4.3) and relied on some efficient third-party libraries, such as ``phyloseq``, ``tidyverse``, ``gridExtra``, and ``magrittr``.

# Results

## Analysis of microbial community structure along with depth and oxygen concentration \label{sec:community}

Motivated by the recent report [@Breitburg2018:declining] regarding the oxygen depletion in the global in the global ocean, we analyze [the contribution of microbes and their existence along .... Fill]. Hereby, we try to understand the compositional complexity of a microbial community across Saanich Inlet samples. For this, we use Shannon's diversity index (SDI), which considers both the species abundances and the total number of distinct species in its diversity estimation. Figures 1(a) and 2(a) depicts Shannon's diversity index for mothur and QIIME2 datasets. Immediately, we observe that SDI values peak at depth $10$ and $100$ before monotonically decreasing at $200$. The SDI values are maximal when the microbes are evenly distributed. Indeed, Figure 3 supports our claim, and we see an uneven distribution of Phylum at $200$ more than at $10$ or $100$ depths. However, the Shannon index values do not capture the number of different species or richness varying across depths. Instead, Figures 1(a) and 2(a) shed some light in this regard. 

Similarly, by analyzing the association between oxygen concentration within a microbial community, we observe, as in Figures 1(b) and 2(b), that SDI values increase when oxygen become more abundant. This is not surprising since the abundance of oxygen indicates..... [FILL]. The boxplots in Figures 1(c) and 2(c) supports this evidence too. Another compleing observation that follows the work in [@Breitburg2018:declining;@Hawley2017:compendium] is the oxygen depletion along the water columns, as illustrated in Figures 1(d) and 2(d). 

[ADD and EDIT]

```{r}
rarefiedMRich <- estimate_richness(rarefiedM, measures = "Shannon")
rarefiedMRichAlpha <- full_join(rownames_to_column(rarefiedMRich), 
                        rownames_to_column(data.frame(sample_data(rarefiedMPer))), 
                        by = "rowname")

p1 <- rarefiedMRichAlpha %>% ggplot() +
  geom_point(aes(x=Depth_m, y=Shannon), size=4, alpha=0.7) +
  geom_smooth(method='loess', aes(x=as.numeric(Depth_m), y=Shannon)) +
  labs(title="Alpha-diversity across depth", 
       y="Shannon's diversity index", x="Depth (m)")

p2 <- rarefiedMRichAlpha %>% ggplot() +
  geom_point(aes(x=O2_uM, y=Shannon), size=4, alpha=0.7) +
  geom_smooth(method='loess', aes(x=as.numeric(O2_uM), y=Shannon)) +
  labs(title="Alpha-diversity across oxygen", 
       y="Shannon's diversity index", x="Oxygen (uM)")

p3 <- rarefiedMRichAlpha %>% 
  mutate(O2_group = ifelse(O2_uM == 0, "anoxic", "oxic")) %>% 
  ggplot() + geom_boxplot(aes(x=O2_group, y=Shannon)) +
  labs(title="Alpha-diversity across oxygen", 
       y="Shannon's diversity index", x="Oxygen (uM)")

p4 <- rarefiedMRichAlpha %>% ggplot() +
  geom_point(aes(x=Depth_m, y=O2_uM), size=4, alpha=0.7) +
  geom_smooth(method='loess', aes(x=as.numeric(Depth_m), y=O2_uM)) +
    labs(title="Oxygen concentration across depth", 
       y="Oxygen  (uM)", x="Depth (m)")

figureM <- ggarrange(p1, p2, p3, p4, ncol = 2, nrow = 2, labels = c("(a)", "(b)", "(c)", "(d)"))
annotate_figure(figureM, bottom = text_grob("Figure 1: Mothur",
                                            face = "bold", 
                                            size = 12))
```


```{r}
rarefiedQRich <- estimate_richness(rarefiedQ, measures = "Shannon")
rarefiedQRichAlpha <- full_join(rownames_to_column(rarefiedQRich), 
                        rownames_to_column(data.frame(sample_data(rarefiedQPer))), by = "rowname")

p1 <- rarefiedQRichAlpha %>% ggplot() +
  geom_point(aes(x=Depth_m, y=Shannon), size=4, alpha=0.7) +
  geom_smooth(method='loess', aes(x=as.numeric(Depth_m), y=Shannon)) +
  labs(title="Alpha-diversity across depth", 
       y="Shannon's diversity index", x="Depth (m)")

p2 <- rarefiedQRichAlpha %>% ggplot() +
  geom_point(aes(x=O2_uM, y=Shannon), size=4, alpha=0.7) +
  geom_smooth(method='loess', aes(x=as.numeric(O2_uM), y=Shannon)) +
  labs(title="Alpha-diversity across oxygen", 
       y="Shannon's diversity index", x="Oxygen (uM)")

p3 <- rarefiedQRichAlpha %>% 
  mutate(O2_group = ifelse(O2_uM == 0, "anoxic", "oxic")) %>% 
  ggplot() + geom_boxplot(aes(x=O2_group, y=Shannon)) +
  labs(title="Alpha-diversity across oxygen", 
       y="Shannon's diversity index", x="Oxygen (uM)")

p4 <- rarefiedQRichAlpha %>% ggplot() +
  geom_point(aes(x=Depth_m, y=O2_uM), size=4, alpha=0.7) +
  geom_smooth(method='loess', aes(x=as.numeric(Depth_m), y=O2_uM)) +
    labs(title="Oxygen concentration across depth", 
       y="Oxygen  (uM)", x="Depth (m)")

figureQ <- ggarrange(p1, p2, p3, p4, ncol = 2, nrow = 2, labels = c("(a)", "(b)", "(c)", "(d)"))
annotate_figure(figureQ, bottom = text_grob("Figure 2: QIIME2",
                                            face = "bold", 
                                            size = 12))
```


```{r fig.width=8}
colourCount <- length(unique(tax_table(secondMTaxa)[, "Phylum"]))

p1 <- plot_bar(secondMTaxa, fill="Phylum") + 
  geom_bar(aes(color = Phylum, fill = Phylum), stat = 'identity', 
           position = 'stack') +
  scale_fill_manual(values = colorRampPalette(brewer.pal(8,
                                                       "Accent"))(colourCount))

annotate_figure(ggarrange(p1, ncol = 1, nrow = 1),
                            bottom = text_grob("Figure 3: Phylum distribution across samples from mothur", 
                                   color = "blue", 
                                   face = "bold", 
                                   size = 12))
```

```{r fig.width=8}
colourCount <- length(unique(tax_table(secondQTaxa)[, "Phylum"]))

p2 <- plot_bar(secondQTaxa, fill="Phylum") + 
  geom_bar(aes(color = Phylum, fill = Phylum), stat = 'identity', 
           position = 'stack')+
  scale_fill_manual(values = colorRampPalette(brewer.pal(8,
                                                       "Accent"))(colourCount))

annotate_figure(ggarrange(p2, ncol = 1, nrow = 1),
                            bottom = text_grob("Figure 4: Phylum distribution across samples from QIIME2", color = "blue", 
                                   face = "bold", size = 12))
```

## Analysis of abundance information of Planctomyces along with depth and/or oxygen concentration \label{sec:OTUabundance}


```{r workingTaxa}
workingMTaxa = subset_taxa(secondMTaxa, Genus == "Planctomyces")
workingQTaxa = subset_taxa(secondQTaxa, Genus == "D_5__Planctomyces")
```

```{r phylum}
phylumM <- get_taxa_unique(physeq = secondMTaxa, taxonomic.rank = "Phylum")
phylumQ <- get_taxa_unique(physeq = secondQTaxa, taxonomic.rank = "Phylum")
phylumMQ <- list(phylumM, phylumQ)
phylumMQ <- data.frame(sapply(phylumMQ, '[', seq(max(sapply(phylumMQ,length))))
                       ,stringsAsFactors=FALSE)
phylumMQ[is.na(phylumMQ)] <- " "
colnames(phylumMQ) <- c("Phylums from Mothur","Phylums from QIIME2")
kable(phylumMQ,caption="Phylums from Mothur vs Phylums from QIIME2")
```

Based on our previous analysis and from Figure $3$, it is clear that the microbial species and their distributions in Saanich inlet samples do indeed vary. Table $1$ summarizes the phyla in both mothur and QIIME2 for Saanich Inlet dataset. We see that the number of hypothetical phyla present in mothur is estimated to be `r length(get_taxa_unique(physeq = secondMTaxa, taxonomic.rank = "Phylum"))`. Throught intesive investigation, we find that the \emph{Planctomycetes} phylum is differentially presented across samples. And, after further exploring its genus distribution, as depicted in Figure 4, we exhibit that \emph{Planctomyces} genus indeed are unevenly distributed across water columns. This is not coincidence because the [Fill based on some paper]. We intitally hypothesize that \emph{Planctomyces} would be represented differently across depths, and for this, we performed regression tests.

```{r fig.height=6.5, fig.width=5}
subMTaxa <- subset_taxa(secondMTaxa, Phylum == "Planctomycetes")
p1 <- plot_bar(subMTaxa, fill="Genus") + 
  geom_bar(aes(color = Genus, fill = Genus), stat = 'identity', 
           position = 'stack')

figureM <- annotate_figure(ggarrange(p1, ncol = 1, nrow = 1, 
                                      widths = 0.7, heights = 0.2),
                            bottom = text_grob("mothur", 
                                   color = "blue", 
                                   face = "bold", 
                                   size = 12))

subQTaxa = subset_taxa(secondQTaxa, Phylum == "D_1__Planctomycetes")
p2 <- plot_bar(subQTaxa, fill="Genus") + 
  geom_bar(aes(color = Genus, fill = Genus), stat = 'identity', 
           position = 'stack')

figureQ <- annotate_figure(ggarrange(p2, ncol = 1, nrow = 1, 
                                      widths = 0.7, heights = 0.2),
                            bottom = text_grob("QIIME2", color = "blue", 
                                   face = "bold", size = 12))

annotate_figure(ggarrange(figureM, figureQ, 
                     ncol = 1, nrow = 2), 
                bottom = text_grob("Figure 5: Genus distribution of Planctomycetes across samples",
                                   face = "bold", 
                                   size = 12))
```


The well known general linear model is employed to recover relationships that might be exhibited between explanatory and target variables. We decompose our hypothesize in a series of experimental tests: i)- first, we investigate the correlation of Planctomyces abundance as a function of depth; and ii)- then, we cross-examine the Planctomyces abundance as a function of oxygen concentration. 

Table 2. shows the results of our test analysis. From the statistical perspective, it can be inferred that there might be no relations with either the depth or the oxyzen. This is becasue the coefficients of depth and oxygen and their p-values were found to be $(-0.0003609,$ p-value $= 0.7385598)$ and $(-0.0002544,$ p-value$=0.7616253)$, respectively, which are not statistically significant at $5\%$ (an arbitrary cutoff). Hence, we might reason that there is a little statistical evidence to support our belief that Planctomyces indeed varies across depth and oxygen. 

Such contradictory conclusion suggests to accept the null hyptothesis that states no intersting patterns exist for Planctomyces. However, the fitting problem associated with the general linear model underestimate the existence of any kind of interesting relationships. Indeed, when we manually inspected the samples, we found that samples from depth $10, 150, 165$ and $200$ do not or are less planctomyces abundant than at $100, 120$, and $135$ depths, which imply that Planctomyces is quite differentially abundant in Saanich Inlet dataset. Perhaps, using more complex models might provide a better predictive analysis; but for now on, we stick with the statistical outputs.  

[WRITE and EDIT]

```{r fig.height=10, fig.width=8}
otu_stats <- data.frame()
lmMTaxa1 <- workingMTaxa %>% tax_glom(taxrank = "Genus") %>% psmelt() %>%
  lm(Abundance ~ Depth_m, .) %>% 
  summary()
lmMTaxa2 <- workingMTaxa %>% tax_glom(taxrank = "Genus") %>% psmelt() %>%
  lm(Abundance ~ O2_uM, .) %>% 
  summary()

lmQTaxa1 <- workingQTaxa %>% tax_glom(taxrank = "Genus") %>% psmelt() %>%
  lm(Abundance ~ Depth_m, .) %>% 
  summary()
lmQTaxa2 <- workingQTaxa %>% tax_glom(taxrank = "Genus") %>% psmelt() %>%
  lm(Abundance ~ O2_uM, .) %>% 
  summary()

otu_stats <- rbind(otu_stats, lmMTaxa1$coefficients["Depth_m",])
otu_stats <- rbind(otu_stats, lmMTaxa2$coefficients["O2_uM",])
otu_stats <- rbind(otu_stats, lmQTaxa1$coefficients["Depth_m",])
otu_stats <- rbind(otu_stats, lmQTaxa2$coefficients["O2_uM",])
tmpNames <- c("Depth (mothur)", "O2_uM (mothur)","Depth (QIIME2)","O2_uM (QIIME2)")
otu_stats <- cbind(tmpNames, otu_stats)
colnames(otu_stats)<- c("Covariates", "Estimate", "Std. Error","t-value","Pr(>|t|)")
kable(otu_stats,caption="Correlation data of OTUs within Planctomyces genus across depth and oxyzen concentration from mothur and QIIME2")
```


```{r}
dfworkingMTaxa <- workingMTaxa %>% psmelt() %>% group_by(Sample) %>% 
  summarize(Abundance_sum=sum(Abundance), Depth_m=mean(Depth_m))
p1 <- ggplot(dfworkingMTaxa) + 
  geom_point(aes(x=Depth_m, y=Abundance_sum), size=5, alpha=0.7) +
  geom_smooth(method="lm", aes(x=as.numeric(Depth_m), y=Abundance_sum)) +
  labs(title="Abundance of Plantomyces across depth") 

dfworkingMTaxa <- workingMTaxa %>% psmelt() %>% group_by(Sample) %>% 
  summarize(Abundance_sum=sum(Abundance), Oxyzen=mean(O2_uM))
p2 <- ggplot(dfworkingMTaxa) +
  geom_point(aes(x=Oxyzen, y=Abundance_sum), size=5, alpha=0.7) +
  geom_smooth(method="lm", aes(x=as.numeric(Oxyzen), y=Abundance_sum)) +
  labs(title="Abundance of Plantomyces with respect to oxyzen concentration")

dfworkingQTaxa <- workingQTaxa %>% psmelt() %>% group_by(Sample) %>% 
  summarize(Abundance_sum=sum(Abundance), Depth_m=mean(Depth_m))
p3 <- ggplot(dfworkingQTaxa) + 
  geom_point(aes(x=Depth_m, y=Abundance_sum), size=5, alpha=0.7) +
  geom_smooth(method="lm", aes(x=as.numeric(Depth_m), y=Abundance_sum)) +
  labs(title="Abundance of Plantomyces across depth") 

dfworkingQTaxa <- workingQTaxa %>% psmelt() %>% group_by(Sample) %>% 
  summarize(Abundance_sum=sum(Abundance), Oxyzen=mean(O2_uM))
p4 <- ggplot(dfworkingQTaxa) +
  geom_point(aes(x=Oxyzen, y=Abundance_sum), size=5, alpha=0.7) +
  geom_smooth(method="lm", aes(x=as.numeric(Oxyzen), y=Abundance_sum)) +
  labs(title="Abundance of Plantomyces with respect to oxyzen concentration")

figureM <- annotate_figure(ggarrange(p1, p2, ncol = 2, nrow = 1, 
                                     labels = c("(a)", "(b)")),
                bottom = text_grob("mothur", 
                                   color = "blue", face = "bold", size = 12))

figureQ <- annotate_figure(ggarrange(p3, p4, ncol = 2, nrow = 1, 
                                     labels = c("(a)", "(b)")),
                bottom = text_grob("QIIME2", 
                                   color = "blue", face = "bold", size = 12))

annotate_figure(ggarrange(figureM, figureQ, 
                     ncol = 1, nrow = 2), 
                bottom = text_grob("Figure 6: Regression analysis of Planctomyces across depth",
                                   face = "bold", 
                                   size = 12))
```


## Estimate richness (number of OTUs/ASVs) for Planctomyces \label{sec:richness}

```{r planctomyces}
suggestedMOTUs <- colnames(otu_table(workingMTaxa))
suggestedQOTUs <- rownames(otu_table(workingQTaxa))

suggestedMQOTUs <- list(suggestedMOTUs, suggestedQOTUs)
suggestedMQOTUs <- data.frame(sapply(suggestedMQOTUs, '[', seq(max(sapply(suggestedMQOTUs,length)))))
colnames(suggestedMQOTUs) <- c("OTUs from Mothur","ASVs from QIIME2")
kable(suggestedMQOTUs,caption="OTUs from Mothur vs ASVs from QIIME2")
```

We explore the diversity of \emph{Planctomyces} across depth. 

[SIMILAR to BOTH PARTS WRITE and EDIT; Consider the abundance information in describing the correlations and shannons diveristy index]



## Interpretation of abundance information of OTUs/ASVs of Planctomyces along with depth and/or oxygen concentration \label{sec:ASVabundances}

```{r}
otu_stats <- data.frame()

for (otu in suggestedMOTUs){
  linear_fit <- workingMTaxa %>% 
    psmelt() %>% 
    filter(OTU==otu) %>% 
    lm(Abundance ~ Depth_m, .) %>% 
    summary()
  otu_stats <- rbind(otu_stats, linear_fit$coefficients["Depth_m",])
}
tmpNames <- paste(suggestedMOTUs," (mothur) ")

for (otu in suggestedQOTUs){
  linear_fit <- workingQTaxa %>% 
    psmelt() %>% 
    filter(OTU==otu) %>% 
    lm(Abundance ~ Depth_m, .) %>% 
    summary()
  otu_stats <- rbind(otu_stats, linear_fit$coefficients["Depth_m",])
}
tmpNames <- c(tmpNames, paste(suggestedQOTUs," (QIIME2) "))
otu_stats <- cbind(tmpNames, otu_stats)
colnames(otu_stats)<- c("Covariates", "Estimate", "Std. Error","t-value","Pr(>|t|)")

kable(otu_stats,caption="Correlation data of OTUs within Planctomyces genus across depth")
```

```{r}
otu_stats <- data.frame()

for (otu in suggestedMOTUs){
  linear_fit <- workingMTaxa %>% 
    psmelt() %>% 
    filter(OTU==otu) %>% 
    lm(Abundance ~ O2_uM, .) %>% 
    summary()
  otu_stats <- rbind(otu_stats, linear_fit$coefficients["O2_uM",])
}
tmpNames <- paste(suggestedMOTUs," (mothur) ")

for (otu in suggestedQOTUs){
  linear_fit <- workingQTaxa %>% 
    psmelt() %>% 
    filter(OTU==otu) %>% 
    lm(Abundance ~ O2_uM, .) %>% 
    summary()
  otu_stats <- rbind(otu_stats, linear_fit$coefficients["O2_uM",])
}
tmpNames <- c(tmpNames, paste(suggestedQOTUs," (QIIME2) "))
otu_stats <- cbind(tmpNames, otu_stats)
colnames(otu_stats)<- c("Covariates", "Estimate", "Std. Error","t-value","Pr(>|t|)")

kable(otu_stats,caption="Correlation data of OTUs within Planctomyces genus across oxygen concentration")
```


```{r fig.height=9}
p1 <- workingMTaxa %>% psmelt() %>% ggplot() +
  geom_point(aes(x=Depth_m, y=Abundance),size=5, alpha=0.7) +
  geom_smooth(method='lm', aes(x=Depth_m, y=Abundance)) +
  facet_wrap(~OTU, scales="free_y") +
  labs(title="Per OTU abundance information")

p2 <- workingMTaxa %>% psmelt() %>% ggplot() +
  geom_point(aes(x=Sample, y=OTU, size=Abundance, color=OTU)) + 
  scale_size_continuous(range = c(0,5)) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(title="Abundance of OTUs across samples")

p3 <- workingQTaxa %>% psmelt() %>% ggplot() +
  geom_point(aes(x=Depth_m, y=Abundance),size=5, alpha=0.7) +
  geom_smooth(method='lm', aes(x=Depth_m, y=Abundance)) +
  facet_wrap(~OTU, scales="free_y") +
  labs(title="Per ASV abundance information")

p4 <- workingQTaxa %>% psmelt() %>% ggplot() +
  geom_point(aes(x=Sample, y=OTU, size=Abundance, color=OTU)) + 
  scale_size_continuous(range = c(0,5)) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(title="Abundance of ASVs across samples")

figureM <- annotate_figure(ggarrange(p1, p2, ncol = 2, nrow = 1, 
                                     labels = c("(a)", "(b)")),
                bottom = text_grob("mothur", 
                                   color = "blue", face = "bold", size = 12))

figureQ <- annotate_figure(ggarrange(p3, p4, ncol = 2, nrow = 1, 
                                     labels = c("(a)", "(b)")),
                bottom = text_grob("QIIME2", 
                                   color = "blue", face = "bold", size = 12))

annotate_figure(ggarrange(figureM, figureQ, ncol = 1, nrow = 2, 
                                     labels = c("(a)", "(b)")),
                bottom = text_grob("Figure 7: Abundance of OTUs/ASVs within Planctomyces genus across depth", face = "bold", size = 12))
```


```{r fig.height=9}
p1 <- workingMTaxa %>% psmelt() %>% ggplot() +
  geom_point(aes(x=O2_uM, y=Abundance),size=5, alpha=0.7) +
  geom_smooth(method='lm', aes(x=O2_uM, y=Abundance)) +
  facet_wrap(~OTU, scales="free_y") +
  labs(title="Per OTU abundance information")

p2 <- workingMTaxa %>% psmelt() %>% ggplot() +
  geom_point(aes(x=Sample, y=OTU, size=O2_uM, color=OTU)) + 
  scale_size_continuous(range = c(0,5)) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(title="Abundance of OTUs across samples")

p3 <- workingQTaxa %>% psmelt() %>% ggplot() +
  geom_point(aes(x=O2_uM, y=Abundance),size=5, alpha=0.7) +
  geom_smooth(method='lm', aes(x=Depth_m, y=Abundance)) +
  facet_wrap(~OTU, scales="free_y") +
  labs(title="Per ASV abundance information")

p4 <- workingQTaxa %>% psmelt() %>% ggplot() +
  geom_point(aes(x=Sample, y=OTU, size=O2_uM, color=OTU)) + 
  scale_size_continuous(range = c(0,5)) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(title="Abundance of ASVs across samples")

figureM <- annotate_figure(ggarrange(p1, p2, ncol = 2, nrow = 1, 
                                     labels = c("(a)", "(b)")),
                bottom = text_grob("mothur", 
                                   color = "blue", face = "bold", size = 12))

figureQ <- annotate_figure(ggarrange(p3, p4, ncol = 2, nrow = 1, 
                                     labels = c("(a)", "(b)")),
                bottom = text_grob("QIIME2", 
                                   color = "blue", face = "bold", size = 12))

annotate_figure(ggarrange(figureM, figureQ, ncol = 1, nrow = 2, 
                                     labels = c("(a)", "(b)")),
                bottom = text_grob("Figure 8: Abundance of OTUs/ASVs within Planctomyces genus across oxygen concentration", 
                                   face = "bold", size = 12))
```


# Discussion

---

# References
